hmrc-mongo-play-27/src/main/scala/uk/gov/hmrc/mongo/play/json/formats/MongoUuidFormats.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mongo.play.json.formats
+
+import org.bson.BsonBinarySubType
+import org.bson.UuidRepresentation
+import org.bson.internal.UuidHelper
+import play.api.libs.json._
+
+import java.util.Base64
+import java.util.UUID
+
+abstract class MongoUuids(uuidRep: UuidRepresentation) {
+  outer =>
+
+  private val encoder = Base64.getEncoder
+  private val decoder = Base64.getDecoder
+
+  private val UuidOldSubtype = BsonBinarySubType.UUID_LEGACY.getValue
+  private val UuidNewSubtype = BsonBinarySubType.UUID_STANDARD.getValue
+
+  private val ChosenSubType =
```

Comment:
would prefer these to not start with capitals

Reply:
That is the Scala [naming convention for constants](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-and-variables) AFAIK but happy to change it